test/suites/peeked-suite.ts

```diff
@@ -0,0 +1,76 @@
+import test = require('tape');
+import {ReadableSignal, Signal} from '../../src';
+import {LeakDetectionSignal} from '../lib/leak-detection-signal';
+import {parentChildSuite} from './parent-child-suite';
+
+export type PeekedSignalCreationFunction = <T>(
+    baseSignal: ReadableSignal<T>,
+    peekaboo: (payload: T) => void,
+) => ReadableSignal<T>;
+export function peekedSuite(prefix: string, createPeekedSignal: PeekedSignalCreationFunction) {
+    parentChildSuite(prefix, () => {
+        const parentSignal = new Signal();
+        const childSignal = createPeekedSignal(parentSignal, _payload => void 0);
+        return { parentSignal, childSignal };
+    });
+
+    test(`${prefix} should be called but not modify the payload`, t => {
```

Comment:
This seems pretty similar to the next test. Are they testing something different? At least it would be good to have different titles.